src/Workspaces/Core/Portable/Remote/RemoteHostClient.cs

```diff
@@ -132,6 +124,18 @@ public void Dispose()
                 _disposed = true;
 
                 OnDisposed();
+
+                GC.SuppressFinalize(this);
+            }
+
+            ~Connection()
+            {
+                // this can happen if someone killed OOP, and when that happen, we don't
+                // want to crash VS. 
```

Comment:
Can this comment be clarified? My reading is that if somebody kills OOP connections might not be disposed and this finalizer will run. But then we run code that will crash VS...?

Reply:
if someone kills OOP. this finalizer (Contract.Fail) can throw and crash VS.